File: JavaScript/1509-Minimum Difference Between Largest and Smallest Value in Three Moves.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minDifference = function(nums) {
    const N = nums.length;
    if (N<=4) return 0;
    let res = Number.POSITIVE_INFINITY;
    nums.sort(function(a,b){return a-b;});
    for (let i=0; i<4; i++){
        res = Math.min(res, nums[N-4+i]-nums[i]);
    }
    return res;
};

File: C++/1509-Minimum Difference Between Largest and Smallest Value in Three Moves.cpp
class Solution {
public:
    int minDifference(vector<int>& nums) {
        int N = nums.size();
        if (N <= 4) return 0;
        sort(nums.begin(), nums.end());
        
        int window = N - 3, mx = nums[0], mn = nums[0];
        int left = 0, res = INT_MAX;
        for (int i=0; i<N; i++) {
            mx = nums[i];
            if (i >= window-1) {
                mn = nums[i-window+1];
            }
            if (i>=window-1) res = min(res, mx-mn);
        }
        return res;
    }
};